### PR TITLE
Use oauth data for permissions

### PR DESCRIPTION
**Pull request by jchristgit**
This PR changes the current permission group assignment implementation to use the permission data provided for the guild from OAuth instead. Since checking against a `RoleMembership` is no longer required, this has the effect that tests data setup gets a lot simpler to write, and the `DISCORD_ADMIN_ROLE_ID` is no longer required. Additionally, it's easier to test locally, since a properly populated instance of statbot is no longer required for the permission setup.

### CODE REVIEW

File: guides/tests/test_admin_user_unowned_guide_interactions.py [hunk 1]
```diff
@@ -4,14 +4,10 @@
...
 from guides.models import Guide
-from stats.models import RoleMembership, Roles, Users as DiscordUser
 from . import INDEX_GUIDE_CONTEXT_NAME
 
 
-@override_settings(
-    DISCORD_ADMIN_ROLE_ID=30,
-    DISCORD_GUILD_ID=55555
-)
+@override_settings(DISCORD_GUILD_ID=55555)
```
Comment (emmiegit) [inline guides/tests/test_admin_user_unowned_guide_interactions.py:10]: `5555` is just a random "guild ID" for testing?

File: guides/tests/test_member_user_single_guide.py [hunk 2]
```diff
@@ -25,28 +24,16 @@ def setUpTestData(cls):
...
-        cls.guild_membership = GuildMembership.objects.create(
-            user=cls.discord_user,
-            guild_id=55555,
-            is_member=True
-        )
         cls.social_account = SocialAccount.objects.create(
             user=cls.member,
             uid=discord_user_id,
-            extra_data={}
+            extra_data={'guilds': [{'id': '55555', 'permissions': 0x63584C0}]}
```
Comment (emmiegit) [inline guides/tests/test_member_user_single_guide.py:36]: Maybe copy this to a constant instead of pasting it everywhere?

File: home/signals.py [hunk 3]
```diff
@@ -4,12 +4,17 @@
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 
-from stats.models import GuildMembership, RoleMembership
+
+ADMIN_PERMISSION_BITFLAGS = 0x8
```
Comment (emmiegit) [inline home/signals.py]: You should add leading zeros that pertain to the width of the flag. I think permission flags are 64-bit, which is admittedly quite long, but you could add enough to at least pad it to some multiple of 4 or 8.

File: home/signals.py [hunk 4]
```diff
@@ -21,32 +26,25 @@ def social_account_post_save(sender, instance, **kwargs):
...
 
-        # Is the newly created user a staff member on the Discord server?
-        is_staff = RoleMembership.objects.filter(
-            user_id=instance.uid,
-            guild_id=settings.DISCORD_GUILD_ID,
-            role_id=settings.DISCORD_ADMIN_ROLE_ID,
-        ).exists()
+        guilds = instance.extra_data['guilds']
+        guild_membership = next(
+            (guild for guild in guilds if guild['id'] == str(settings.DISCORD_GUILD_ID)),
```
Comment (emmiegit) [inline home/signals.py:31]: Alternatively,
```py
filter(lambda g: g['id'] == str(settings.DISCORD_GUILD_ID), guilds),
```

File: guides/tests/test_admin_user_unowned_guide_interactions.py (hunk 1 above)
Comment (jchristgit) [reply guides/tests/test_admin_user_unowned_guide_interactions.py:10]: Yes

File: guides/tests/test_member_user_single_guide.py (hunk 2 above)
Comment (jchristgit) [reply guides/tests/test_member_user_single_guide.py:36]: This is used once per file and the permissions can be fairly random as long as they do not have the admin bitflag set, which is why I just put it in like this.

File: home/signals.py (hunk 4 above)
Comment (jchristgit) [reply home/signals.py:31]: I think this way is more readable.

File: home/signals.py (hunk 3 above)
Comment (jchristgit) [reply home/signals.py]: Done

**Review by emmiegit — APPROVED** [review]

**Pull request merged**